### ISSUE TITLE
Fails to decode .gif

### ISSUE THREAD

**Issue by lilith** [image-rs/image-gif#138]
In 0.11.4, I can't get this to decode: I get DecodingFormatError { underlying: "unexpected EOF" }
It works in Chrome/Win/Brave.
https://user-images.githubusercontent.com/657201/139770827-18e25c4e-eb0a-4058-ba48-ddc3849090ee.gif

**Comment by HeroicKatora** [comment]
This image is missing a Trailer block to end its data stream. The netbpm tools agree with this:

```
$ giftopnm /tmp/139770827-18e25c4e-eb0a-4058-ba48-ddc3849090ee.gif > /dev/null 
giftopnm: Error accessing Image 67 of stream.  File read error where start of image descriptor or end of GIF expected.  End of file encountered
```

Frames before this are handled fine but eagerly decoding the complete file will surface the error. Is there any other error strategy that we could do here instead to recover?

**Comment by mpwsh** [comment]
Im having the same problem. Now gif won't decode anything, it fails with `unexpected EOF`.
Only thing i did was update from 0.11.4 to 0.12.0 and move to toolchain 1.65.0.


```bash
wget "https://ipfs.io/ipfs/bafybeihqwohkbcwf2egmwd4lnriaa63xc3eqfbyb4i4qftjikyp5wtwlqi/5220.gif" -O claydino.gif
giftopnm claydino.gif  > /dev/null
```

With gif lib
```bash
[2022-11-29T16:34:13Z INFO  imgopt::utils] Opening file:: storage/base/ipfs/bafybeihqwohkbcwf2egmwd4lnriaa63xc3eqfbyb4i4qftjikyp5wtwlqi/5220.gif
[2022-11-29T16:34:13Z INFO  imgopt::utils] reading image from file: took 638 µs
[2022-11-29T16:34:13Z WARN  imgopt::img] Resizing gif
[2022-11-29T16:34:13Z WARN  imgopt::img] starting read_info
[2022-11-29T16:34:13Z ERROR imgopt::img] unexpected EOF
```

Is there something we can do to handle the 'unexpected EOF' error?

**Comment by fintelia** [comment]
Yeah, read_to_end leaves the file descriptor pointing at the end of the file. If the decoder later tries to read even a single byte it'll get back an error that there's no more data to read

**Comment by lilith** [comment]
I call read_info() first. Is there a way to tolerate GIF files without a trailer block the way a browser will?
